registry/remote/example_test.go

```diff
@@ -0,0 +1,326 @@
+	desc := ocispec.Descriptor{                                                       // Assemble a descriptor
+		MediaType: mediaType,                 // Set mediatype
+		Digest:    digest.FromBytes(content), // Calculate digest
+		Size:      int64(len(content)),       // Include content size
+	}
+	err = repo.Push(ctx, desc, bytes.NewReader(content)) // Push the blob
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	fmt.Println("Push finished")
+	// Output:
+	// Push finished
+}
+
+// ExampleRepository_Resolve_byTag gives example snippets for resolving a tag.
+func ExampleRepository_Resolve_byTag() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	tag := "latest"
+	descriptor, err := repo.Resolve(ctx, tag) // Resolve digest to the descriptor
+	if err != nil {
+		panic(err) // Handle error
+	}
+	fmt.Println(descriptor.MediaType)
+	fmt.Println(descriptor.Digest)
+	fmt.Println(descriptor.Size)
+
+	// Output:
+	// application/vnd.oci.image.layer.v1.tar
+	// sha256:aafc6b9fa2094cbfb97eca0355105b9e8f5dfa1a4b3dbe9375a30b836f6db5ec
+	// 20
+}
+
+// ExampleRepository_Resolve_byDigest gives example snippets for resolving a digest.
+func ExampleRepository_Resolve_byDigest() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+	digest := "sha256:aafc6b9fa2094cbfb97eca0355105b9e8f5dfa1a4b3dbe9375a30b836f6db5ec"
+	descriptor, err := repo.Resolve(ctx, digest) // Resolve digest to the descriptor
+	if err != nil {
+		panic(err) // Handle error
+	}
+	fmt.Println(descriptor.MediaType)
+	fmt.Println(descriptor.Digest)
+	fmt.Println(descriptor.Size)
+
+	// Output:
+	// application/vnd.oci.image.layer.v1.tar
+	// sha256:aafc6b9fa2094cbfb97eca0355105b9e8f5dfa1a4b3dbe9375a30b836f6db5ec
+	// 20
+}
+
+// ExampleRepository_Fetch_byTag gives example snippets for downloading a blob by tag.
+func ExampleRepository_Fetch_byTag() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	tag := "latest"
+	descriptor, err := repo.Resolve(ctx, tag) // First resolve the tag to the descriptor
+	if err != nil {
+		panic(err) // Handle error
+	}
+	r, err := repo.Fetch(ctx, descriptor) // Fetch the blob from the repository
+	if err != nil {
+		panic(err) // Handle error
+	}
+	defer r.Close() // don't forget to close
+	pulledBlob, err := io.ReadAll(r)
```

Comment:
Use `content.FetchAll()` for validation

Reply:
Should we use it as validation here?
I can add a new example for `FetchAll()` if that is what you want

Reply:
Resolved. Added digest validation here and new examples for content.FetchAll()